hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -199,18 +199,29 @@ protected List< OmKeyLocationInfo > allocateBlock(ScmClient scmClient,
     List<OmKeyLocationInfo> locationInfos = new ArrayList<>(numBlocks);
     String remoteUser = getRemoteUser().getShortUserName();
     List<AllocatedBlock> allocatedBlocks;
-    try {
-      allocatedBlocks = scmClient.getBlockClient()
-          .allocateBlock(scmBlockSize, numBlocks, replicationConfig, serviceID,
-              excludeList, clientMachine);
-    } catch (SCMException ex) {
-      omMetrics.incNumBlockAllocateCallFails();
-      if (ex.getResult()
-          .equals(SCMException.ResultCodes.SAFE_MODE_EXCEPTION)) {
-        throw new OMException(ex.getMessage(),
-            OMException.ResultCodes.SCM_IN_SAFE_MODE);
+    int retryCount = 5;
+    while (true) {
+      try {
+        allocatedBlocks = scmClient.getBlockClient()
+            .allocateBlock(scmBlockSize, numBlocks, replicationConfig, serviceID,
+                excludeList, clientMachine);
+      } catch (SCMException ex) {
+        omMetrics.incNumBlockAllocateCallFails();
+        if (ex.getResult().equals(SCMException.ResultCodes.SAFE_MODE_EXCEPTION) && retryCount > 0) {
+          retryCount--;
+          // SCM is in safe mode, retry again
```

Comment:
Maybe add a debug log saying " allocate block failed as SCM is in safe mode , x retries remaining"